File: 2-1.py
#!/usr/bin/env python

num_valid = 0

f = open('2.input')
for line in f.readlines():
    count, allowed, passwd = line.split(' ')
    min, max = count.split('-')
    min = int(min)
    max = int(max)
    c = allowed[0]
    current_count = 0
    for x in passwd:
        if x == c:
            current_count += 1
    if current_count >= min and current_count <= max:
        num_valid +=1

print(num_valid)

File: 2-1.rb
num_valid = 0

File.readlines('2.input').each do |row|
  count, allowed, pass = row.split(' ')
  min = count.split('-').first.to_i
  max = count.split('-').last.to_i
  c = allowed[0]
  current_count = 0
  pass.split('').each do |x|
    if x == c then
      current_count += 1
    end
  end
  if current_count >= min and current_count <= max then
    num_valid += 1
  end
end

print num_valid, "\n"
